core/src/test/java/io/dekorate/utils/BeansTest.java

```diff
@@ -0,0 +1,172 @@
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * 
+**/
+
+package io.dekorate.utils;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.assertj.core.util.Arrays;
+import org.junit.jupiter.api.Test;
+
+public class BeansTest {
+
+  private static class Person {
+    private final String name;
+    private final String address;
+    private final String phone;
+
+    public Person() {
+      this(null, null, null);
+    }
+
+    public Person(String name, String address, String phone) {
+      this.name = name;
+      this.address = address;
+      this.phone = phone;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public String getAddress() {
+      return address;
+    }
+
+    public String getPhone() {
+      return phone;
+    }
+  }
+
+  private static class Team {
+    private final String name;
+    private final Person captain;
+    private final List<Person> members;
+
+    public Team() {
+      this(null, null, null);
+    }
+
+    public Team(String name, Person captain, List<Person> members) {
+      this.name = name;
+      this.captain = captain;
+      this.members = members;
+    }
+
+    public String getName() {
+      return name;
+    }
+
+    public Person getCaptain() {
+      return captain;
+    }
+
+    public List<Person> getMembers() {
+      return members;
+    }
+
+  }
+     
+
+  @Test
+  public void shouldCombineSimpleBeans() throws Exception {
+    Person person1 = new Person("Jim", null, null);
+    Person person2 = new Person(null, "Some place", null);
+    Person combined = Beans.combine(person1, person2);
+    assertNotNull(combined);
+    assertEquals("Jim", combined.getName());
+    assertEquals("Some place", combined.getAddress());
+  }
+
+
+  @Test
+  public void shouldCombineSynteticObjects() throws Exception {
+    Person captain = new Person("Jim", null, null);
+    Person member1 = new Person("Joe", null, null);
+    Team team1 = new Team("A", captain, null);
+    List<Person> members = new ArrayList<>();
+    members.add(member1);
+    Team team2 = new Team("A", null, members);
+    Team combined = Beans.combine(team1, team2);
+    assertNotNull(combined);
+    assertEquals("A", combined.getName());
+    assertNotNull(combined.getMembers());
```

Comment:
This is really minor/secondary but I would add a check about the captain of combined because it helps to validate for my (and maybe is the same for others) assumptions about the code : `assertEquals(captain, combined.getCaptain());`